internal/querycoord/task.go

```diff
@@ -2252,36 +2252,38 @@ func (lbt *loadBalanceTask) postExecute(context.Context) error {
 
 func (lbt *loadBalanceTask) globalPostExecute(ctx context.Context) error {
 	if len(lbt.getChildTask()) > 0 {
+		replicas := make(map[UniqueID]*milvuspb.ReplicaInfo)
+		segments := make(map[UniqueID]*querypb.SegmentInfo)
+
+		for _, id := range lbt.SourceNodeIDs {
+			for _, segment := range lbt.meta.getSegmentInfosByNode(id) {
+				segments[segment.SegmentID] = segment
+			}
+
+			nodeReplicas, err := lbt.meta.getReplicasByNodeID(id)
+			if err != nil {
+				log.Warn("failed to get replicas for removing offline querynode from it",
+					zap.Int64("querynodeID", id),
+					zap.Error(err))
+			}
```

Comment:
Assuming the `nodeReplicas` is nil in error branch, the logic works here. Just two things:
1. Continue the loop to explicitly skip the following operations will be better
2. If the source node does not has replica attached, is this load balance task valid anymore?

Reply:
1. Agree
2. Source nodes can be only the nodes in replica